hs_core/hydro_realtime_signal_processor.py

```diff
@@ -59,15 +59,57 @@ def handle_save(self, sender, instance, **kwargs):
+                pass
+        
+        elif isinstance(instance, AbstractMetaDataElement):
+            try: 
+                # resolve the BaseResource corresponding to the metadata element. 
+                # fields used here are the union of all fields in the metadata element
+                # some of these are deprecated, but in the interest of time, 
+                # I included them now and will delete later. 
+                newinstance = BaseResource.objects.get(Q(metadata__title=instance) |
+                                                       Q(metadata__description=instance) |
+                                                       Q(metadata__creators=instance) |
+                                                       Q(metadata__contributors=instance) |
+                                                       Q(metadata__citation=instance) |
+                                                       Q(metadata__dates=instance) |
+                                                       Q(metadata__coverages=instance) |
+                                                       Q(metadata__formats=instance) |
+                                                       Q(metadata__identifiers=instance) |
+                                                       Q(metadata__language=instance) |
+                                                       Q(metadata__subjects=instance) |
+                                                       Q(metadata__sources=instance) |
+                                                       Q(metadata__relations=instance) |
+                                                       Q(metadata__rights=instance) |
+                                                       Q(metadata__type=instance) |
+                                                       Q(metadata__publisher=instance) |
+                                                       Q(metadata__funding_agencies=instance) |
+                                                       Q(metadata__variables=instance) |
+                                                       Q(metadata__sites=instance) |
+                                                       Q(metadata__methods=instance) |
+                                                       Q(metadata__quality_levels=instance) |
+                                                       Q(metadata__datasources=instance) |
+                                                       Q(metadata__time_series_results=instance) |
+                                                       Q(metadata__variables=instance) |
+                                                       Q(metadata__fieldinformations=instance))
+                self.handle_save(BaseResource, newinstance)
+            except Exception:  # don't report missing resource 
+                pass
 
+        else:  # could be extended metadata element
+            try: 
+                newinstance = BaseResource.objects.get(extra_metadata=instance)
```

Comment:
There are other hidden metadata elements on aggregation metadata models, this is a comment to note it, not expecting any action at this time.